### DISCUSSION TITLE
Comparison of Shell Tensor/Scalar data between LS-Dyna D3plot and OpenRadioss Anim

### DISCUSSION THREAD

**Discussion by DavidR-Vortex** [OpenRadioss/OpenRadioss#1214] [Ideas]
Hi,

I have run the Camry model in both LS-Dyna and OpenRadioss and done a comparison on the arrays available. It's possible that I've not been able to unpack every array name, if so apologies.

I have some code currently using d3plots to extract data from, what I would like to do is find a clean way to do the same for the OpenRadioss Anim binaries.

Some of the things I extract are:

- 'part_mass'
- 'element_shell_thickness'

I'm seemingly unable to extract similar data from the Radioss Anim 

I'd like to discuss the best way to go about resolving this.

Could we:

 A:

- Update the defaults in OpenRadioss such that shell thickness and mass are output as Dyna does in it's current form

B:

- Change values in the current Dyna deck to trigger OpenRadioss into outputting these values when the Dyna deck gets mapped

C:

- Add additional Open Radioss ANIM commands with the include_dyna to output these.

The full list of array names that I have been able to extract are below, some like the subset category have another level:

OpenRadioss

**['element_beam_node_indexes', 'element_shell_node_indexes', 'element_solid_node_indexes', 'node_ids', 'node_time_step', 'node_mass_change', 'node_velocity', 'node_displacement', 'node_acceleration', 'node_contact_forces', 'node_contact_pressure___normal', 'node_contact_pressure___tangent', 'node_tied_contact_forces', 'element_beam_ids', 'element_shell_ids', 'element_solid_ids', 'element_beam_part_indexes', 'element_beam_part_names', 'element_shell_part_indexes', 'element_shell_part_names', 'element_solid_part_indexes', 'element_solid_part_names', 'element_beam_is_alive', 'element_shell_is_alive', 'element_solid_is_alive', 'element_beam_plastic_strain', 'element_beam_specific_energy', 'element_shell_time_step', 'element_shell_mass_change', 'element_shell_plastic_strain', 'element_shell_specific_energy', 'element_shell_stress_(upper)', 'element_shell_stress_(lower)', 'element_solid_plastic_strain', 'element_solid_specific_energy', '2:rbodies___rbe2___rbe3_model2004501', '3:rwalls_model2___rbe3_model2004501', '1:global_model2___rbe3_model2004501', 'material_text_a', 'material_type_a', 'properties_text_a', 'properties_type_a']**

LS-Dyna

**['node_coordinates', 'element_solid_node_indexes', 'element_solid_part_indexes', 'element_tshell_node_indexes', 'element_tshell_part_indexes', 'element_beam_part_indexes', 'element_beam_node_indexes', 'element_shell_node_indexes', 'element_shell_part_indexes', 'node_ids', 'element_solid_ids', 'element_beam_ids', 'element_shell_ids', 'element_tshell_ids', 'part_ids', 'part_ids_unordered', 'part_ids_cross_references', 'part_titles_ids', 'part_titles', 'timesteps', 'global_kinetic_energy', 'global_internal_energy', 'global_total_energy', 'global_velocity', 'part_internal_energy', 'part_kinetic_energy', 'part_velocity', 'part_mass', 'part_hourglass_energy', 'rigid_wall_force', 'node_displacement', 'node_velocity', 'node_acceleration', 'element_solid_stress', 'element_solid_effective_plastic_strain', 'element_beam_axial_force', 'element_beam_shear_force', 'element_beam_bending_moment', 'element_beam_torsion_moment', 'element_shell_effective_plastic_strain', 'element_shell_thickness', 'element_shell_unknown_variables', 'element_shell_internal_energy', 'element_solid_is_alive', 'element_shell_is_alive', 'element_beam_is_alive']**

Many thanks,
David

**Comment by PaulAltair** [comment]
you can add shell thickness output by adding requests /ANIM/SHELL/THIC (additionally /ANIM/SHELL/THIN gives % thinning) manually to the 0001.rad generated by the starter executable (before running the engine) I will request that this be added by default

Part mass is not an animation output in OpenRadioss, but it can be requested as a time history output in /TH/PART

**Reply by DavidR-Vortex** [reply]
Thank you Paul, that additional database request appears to be working for me.

I also requested for /ANIM/SHELL/DENS and got an array of zeros out. I'm guessing that this is a bug?

Kind regards,
David

**Comment by sebastienVilleneuve** [comment]
Hi @DavidR-Vortex 

you said : 'I have some code currently using d3plots to extract data from'
Do you have also code that is able to write d3plot ?

Thanks,
Sébastien

**Reply by DavidR-Vortex** [reply]
Hi @sebastienVilleneuve

This library has that functionality:

https://open-lasso-python.github.io/lasso-python/dyna/D3plot/

The pro features are now part of the standard package but I've not used the functionality myself,

Hope it helps,

EDIT

@sebastienVilleneuve is the plan to create a conversion tool for Anim to D3plot? If so I already have a Python class for reading Anim files. Having such a tool would be useful to me & I would be happy to upload what I have if it helps in the development.

Kind regards,
David

**Reply by Popular1319** [reply]
Hi @DavidR-Vortex, do you have the script to write d3plot directly from anim files, can you help me with this conversion process. 
Thanks in advance
